Improve logging of violations in test mode

EndangeredMassa opened the pull request:
This PR extracts the violation formatting logic into a util class that can be used in both component audit and test audit scenarios.

It also changes the violationsHelper so that it works more like a singleton and is used throughout the project as a module instead of a global. It's still exposed as a global in the initializer, though. This also makes it work properly in test mode.

---

Here's a before/after of using ember-a11y-testing in another project's test suite:

Before:

![](https://dha4w82d62smt.cloudfront.net/items/0v2f0v0i1W0m0Q3f3c3A/Image%202018-05-17%20at%2012.43.28%20PM.png?X-CloudApp-Visitor-Id=221f7ce2733e98e780fb509218a58153&v=70c5437d)

After:

![](https://user-images.githubusercontent.com/41545/40194590-954f9eb8-59d0-11e8-877e-51ef8ff3b7b6.png)

---

Here's a screenshot of development mode to show that it still works fine:

![](https://dha4w82d62smt.cloudfront.net/items/0O1Q1i1J433T2o0p1t2m/Image%202018-05-17%20at%2012.46.08%20PM.png?X-CloudApp-Visitor-Id=221f7ce2733e98e780fb509218a58153&v=7d0e1e1f)

dbollinger commented:
Looks good-- you are a saint! Any reason not to continue outputting the total number of violations?

EndangeredMassa commented:
Added a violation count to the error message. It now looks like:

```
[serious]: Elements must have sufficient color contrast 
Violated 5 times. Offending nodes are: 
<span class="c-radio-track__item-node">None</span>
<span class="c-radio-track__item-node">Level 1</span>
<span class="c-radio-track__item-node">Level 2</span>
<span class="c-radio-track__item-node">Level 3</span>
<p data-test-selector="labeless-text-input" id="violations__low-contrast-text" class="p-violations__grid-item-content--low-contrast-text c-paragraph ember-view">          Swoooosh
</p>
https://dequeuniversity.com/rules/axe/2.6/color-contrast?application=axeAPI
{object}
```

I also changed `Offending markup is` to `Offending nodes are` because what follows is a list of every node that violated that rule.

JoaoGFarias commented:
Thank you! This improvement makes checking and debugging a11y in Ember apps so much faster!
